### DISCUSSION TITLE
Access private MkDocs after logging in

### DISCUSSION THREAD

**Discussion by dasrocky** [squidfunk/mkdocs-material#5875] [Q&A] [answered]
This might sound stupid question, i am trying to deploy Material for MkDocs in private repo and it can be accessible by limited authorized people either through google login screen or some other way. I am still new to mkdocs, i did try it out.

Is this something possible? Any suggestion would be welcome, at least it will give me a path to start.

**Comment by squidfunk** [comment, marked as answer]
Not with MkDocs in particular, but you can use any authentication system in front of your site, because the site is just static HTML. One of the simplest ideas is to use [nginx](https://www.nginx.com/) for hosting and [.htpasswd](https://docs.nginx.com/nginx/admin-guide/security-controls/configuring-http-basic-authentication/) for access control

**Comment by dasrocky** [comment]
@squidfunk Your suggestion is one way of doing it, I took a little  different approach, used fastapi for auth and then redirection to mkdocs url. Haven't fully implemented it yet. 

I have another question, is it possible to set limit to access to some page based on mkdocs url (mkdocs nav)? i hope my question makes sense

**Reply by alexvoss** [reply]
@dasrocky this will depend on what you use for hosting. With Nginx or Apache you would set access restrictions in the server configuration based on the URL. The question of authentication/authorization is really orthogonal to the production of the content with MkDocs&Material.

I assume you have chosen fastapi because you are developing some aspect of the site with it? If you just picked it as a web server then I suggest you have a look at Nginx as this is a better solution for hosting content.

**Comment by humitos** [comment]
Hi! There is a software called `staticrypt` that you may find useful for this purpose.

I've created an example using "Materia for MkDocs" to generate the documentation, `staticrypt` to encrypt it under a password and Read the Docs to build/host/serve them. Take a look at it at https://test-builds.readthedocs.io/en/staticrypt/ (password is `humitos`)

The config file that generates that site is at https://github.com/readthedocs/test-builds/blob/staticrypt/.readthedocs.yaml

I hope this helps for your use case 👍🏼

**Comment by ben519** [comment]
I used Netlify Identity + JavaScript to gate content in [my site](https://www.practiceprobs.com/). Admittedly, it was kind of tricky to implement, so I built [Scipress.io](https://www.scipress.io/) as an alternative.

**Comment by gabrielbdornas** [comment]
I encountered the same challenge when trying to protect an MkDocs Material static site with authentication. After some research and testing, I found a way to solve this using [OAuth2 Proxy](https://oauth2-proxy.github.io/oauth2-proxy/) with GitHub authentication.  

I've documented my approach in this repository: **[meadapt/login-static-site](https://github.com/meadapt/login-static-site)**.  

### Key Points of My Solution:
- The static site is built using MkDocs Material and deployed using GitHub Actions and [Render](https://render.com/).
- OAuth2 Proxy is configured to require GitHub login, ensuring only authorized users can access the site, but other OAuth providers, like Google, could be used. 
- Authentication is enforced using an email whitelist, where only specific GitHub users can log in.
- The entire process is deployed using a Render web service to handle authentication and serve the protected content.

If you’re looking for a way to secure your MkDocs Material site with GitHub authentication, check out the [repository](https://github.com/meadapt/login-static-site). The `README.md` provides detailed setup instructions, including how to create an OAuth App on GitHub, configure environment variables, and deploy everything on Render.  

Let me know if you have any questions! 🚀

**This is, also, another oportunity to thanks @squidfunk for his brilhant work. The [mkdocs-material](https://github.com/squidfunk/mkdocs-material) is one of the best GitHub repos in my modest opinion!**  :muscle:
